crates/bevy_ecs/src/schedule/stage.rs

```diff
@@ -563,6 +572,20 @@ impl SystemStage {
         }
     }
 
+    fn check_uses_resource(&self, resource_id: ComponentId, world: &World) {
```

Comment:
I'm assuming this new constraint is purely to enforce "best practices" not correctness? Presumably because devs shouldn't be doing non-extract work in the extract step?

Reply:
Yes and no

It is to enforce best practice. However, much more importantly it makes porting code to this version significantly more tractable.

It's otherwise nearly impossible to find all Extract systems and confirm you have updated them.